src/cli/repl/commands/lineage.ts

```diff
@@ -0,0 +1,77 @@
+import type { ReplCommand } from './main'
+import { PipelineExecutor } from '../../../core/pipeline-executor'
+import { DEFAULT_DATAFLOW_PIPELINE } from '../../../core/steps/pipeline/default-pipelines'
+import type { RShell } from '../../../r-bridge/shell'
+import { requestFromInput } from '../../../r-bridge/retriever'
+import type { SingleSlicingCriterion } from '../../../slicing/criterion/parse'
+import { slicingCriterionToId } from '../../../slicing/criterion/parse'
+import type { NodeId } from '../../../r-bridge/lang-4.x/ast/model/processing/node-id'
+import type { OutgoingEdges } from '../../../dataflow/graph/graph'
+import type { DataflowGraphEdge } from '../../../dataflow/graph/edge'
+import { edgeIncludesType, EdgeType } from '../../../dataflow/graph/edge'
+import type { DataflowGraphVertexInfo } from '../../../dataflow/graph/vertex'
+import type { DataflowInformation } from '../../../dataflow/info'
+import type { NormalizedAst } from '../../../r-bridge/lang-4.x/ast/model/processing/decorate'
+
+function splitAt(str: string, idx: number) {
+	return [str.slice(0, idx), str.slice(idx)]
+}
+
+async function getDfg(shell: RShell, remainingLine: string) {
+	return await new PipelineExecutor(DEFAULT_DATAFLOW_PIPELINE, {
+		shell,
+		request: requestFromInput(remainingLine.trim())
+	}).allRemainingSteps()
+}
+
+function filterRelevantEdges(edge: DataflowGraphEdge) {
+	return edgeIncludesType(EdgeType.DefinedBy | EdgeType.DefinedByOnCall | EdgeType.Returns | EdgeType.Reads, edge.types)
+}
+
+function pushRelevantEdges(queue: [NodeId, DataflowGraphEdge][], outgoingEdges: OutgoingEdges) {
+	queue.push(...[...outgoingEdges].filter(([_, edge]) => filterRelevantEdges(edge)))
+	return queue
```

Comment:
with the push modifying the queue in line, returning it additionally with two spreads in the sources may give the wrong impression?

Reply:
It was handy for line 47
https://github.com/flowr-analysis/flowr/blob/6ae488574a40dac84be1647dbe3a4b4f935d89e5/src/cli/repl/commands/lineage.ts#L47
But I get your point. I'll update it.